modernizer-maven-plugin/src/test/java/org/gaul/modernizer_maven_plugin/ModernizerSuppressionsEndToEndTest.java

```diff
@@ -0,0 +1,285 @@
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreConstructorTestClass;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreGenericClass;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreGenericClassConstructorTestClass;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreMethodInGenericClassTest;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreMethodReturningArrayClassTest;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreMethodReturningArrayPrimitiveTypeClassTest;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreMethodReturningDeclaredTypeClassTest;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreMethodReturningGenericTypeClassTest;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreMethodReturningPrimitiveTypeClassTest;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreMethodWithArrayTypeParametersTestClass;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreMethodWithDeclaredTypeParametersTestClass;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreMethodWithEmptyParametersTestClass;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreMethodWithGenericTypeParametersTestClass;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreMethodWithPrimitiveAndGenericTypeParametersTestClass;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreMethodWithPrimitiveTypeParametersTestClass;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreMethodWithVoidParameterTestClass;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreOverloadedMethodInGenericClassTest;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreOverloadedMethodTestClass;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreStaticOuterClassConstructor.IgnoreStaticInnerClassConstructor;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.objectweb.asm.ClassReader;
+
+public final class ModernizerSuppressionsEndToEndTest {
+    private static Map<String, Violation> violations;
+    private static final Collection<String> NO_EXCLUSIONS =
+        Collections.<String>emptySet();
+    private static final Collection<Pattern> NO_EXCLUSION_PATTERNS =
+        Collections.<Pattern>emptySet();
+    private static final Collection<String> NO_IGNORED_PACKAGES =
+        Collections.<String>emptySet();
+    private static final Collection<String> NO_IGNORED_METHODS =
+        Collections.<String>emptySet();
+    private static Set<Pattern> ignoreClasses = new HashSet<Pattern>();
+    private static Set<String> ignoreMethods = new HashSet<String>();
```

Comment:
I think it wasn't clear. Do `private static Set<Pattern> ignoreClasses;` here and `ignoreMethods = new HashSet<String>;` in `setUp`